client/src/components/LibraryFolder/TopToolbar/FolderTopBar.vue

```diff
@@ -0,0 +1,382 @@
+            auto: {
+                id: "auto",
+                text: "Auto-detect",
+                description:
+                    "This system will try to detect the file type automatically." +
+                    " If your file is not detected properly as one of the known formats," +
+                    " it most likely means that it has some format problems (e.g., different" +
+                    " number of columns on different rows). You can still coerce the system" +
+                    " to set your data to the format you think it should be." +
+                    " You can also upload compressed files, which will automatically be decompressed.",
+            },
+        };
+    },
+    created() {
+        const Galaxy = getGalaxyInstance();
+        this.is_admin = Galaxy.user.attributes.is_admin;
+        this.user_library_import_dir = Galaxy.config.user_library_import_dir;
+        this.library_import_dir = Galaxy.config.library_import_dir;
+        this.allow_library_path_paste = Galaxy.config.allow_library_path_paste;
+        if (
+            this.user_library_import_dir !== null ||
+            this.allow_library_path_paste !== false ||
+            this.library_import_dir !== null
+        ) {
+            this.multiple_add_dataset_options = true;
+        }
+        const contains_file_or_folder = this.folderContents.find((el) => el.type === "folder" || el.type === "file");
+
+        // logic from legacy code
+        if (contains_file_or_folder) {
+            if (Galaxy.user) {
+                this.dataset_manipulation = true;
+                if (!Galaxy.user.isAnonymous()) {
+                    this.logged_dataset_manipulation = true;
+                }
+            }
+        }
+        this.fetchExtAndGenomes();
+    },
+    mounted() {
+        new mod_path_bar.PathBar({
+            full_path: this.metadata.full_path,
+            id: this.folder_id,
+            parent_library_id: this.metadata.parent_library_id,
+        });
+    },
+    computed: {
+        allDatasets: function () {
+            return this.folderContents.filter((element) => element.type === "file");
+        },
+    },
+    methods: {
+        updateSearch: function (value) {
+            this.$emit("updateSearch", value);
+        },
+        deleteSelected: function () {
+            deleteSelectedItems(
+                this.selected,
+                (deletedItem) => this.$emit("deleteFromTable", deletedItem),
+                () => this.$emit("refreshTable")
+            );
+        },
+        newFolder() {
+            this.folderContents.unshift({
+                editMode: true,
+                isNewFolder: true,
+                type: "folder",
+                name: "",
+                description: "",
+            });
+            this.$emit("refreshTable");
+        },
+        downloadData(format) {
+            const { datasets, folders } = this.findCheckedItems();
+            if (this.selected.length === 0) {
```

Comment:
Since there is no risk of implicit conversion here `==` is sufficient.

Reply:
Actually... why do we care whether there is a risk of implicit conversion or not here? Why do we always have to always use `==`, if there is no risk?

Reply:
We should lean *towards* using explicit equality `===` where we can, not relying on whether or not `==` is sufficient.  `===` is the right thing to use here.

Reply:
I agree with @dannon, `===` should be preferable